src/api/client/Network.ts

```diff
@@ -549,4 +549,29 @@ export class Network {
   public async getMiddlewareMetadata(): Promise<MiddlewareMetadata | null> {
     return this.context.getMiddlewareMetadata();
   }
+
+  /**
+   * Check if the middleware is up to date with the chain.
+   *
+   * @note Since the middleware can take some time in processing large blocks,
+   * the function returns true if the number of processed blocks by the middleware doesn't lag the finalized chain blocks by more than 20.
+   * More details about the last processed block from middleware can be fetched using the `getMiddlewareMetadata` method.
+   *
+   * @note uses the middleware V2
+   */
+  public async isMiddlewareSynced(): Promise<boolean> {
```

Comment:
What about returning how many blocks middleware is laging, instead of returning a true/false value here?

I don't think we can find one value that is right for every application